src/externalized/army/ArmyCompositionModel.cc

```diff
@@ -0,0 +1,88 @@
+#include "ArmyCompositionModel.h"
+
+
+ArmyCompositionModel::ArmyCompositionModel(uint8_t compositionId_, const char* name_, int8_t priority_,
+		int8_t adminPercentage_, int8_t elitePercentage_, int8_t troopPercentage_,
+		int8_t desiredPopulation_, int8_t startPopulation_) :
+			compositionId(compositionId_), name(name_), priority(priority_),
+			adminPercentage(adminPercentage_), elitePercentage(elitePercentage_), troopPercentage(troopPercentage_),
+			desiredPopulation(desiredPopulation_), startPopulation(startPopulation_) {}
+
+ARMY_COMPOSITION ArmyCompositionModel::toArmyComposition() const
+{
+	return {
+		compositionId,
+		priority,
+		elitePercentage,
+		troopPercentage,
+		adminPercentage,
+		desiredPopulation,
+		startPopulation,
+		{ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }
```

Comment:
These zeroes are just relics for save game compatibility? We should document it, since it looks odd in all these models.

Reply:
I avoid list initialization like this because it's hard to search and hides simple mistakes like using the wrong order or missing a field (can result in uninitialized memory).
It's also a pain to change when the structure changes.

This test shows list initialization forms that are known to initialize all the memory in our CI compilers (I recommend zero-initialize forms):
https://github.com/ja2-stracciatella/ja2-stracciatella/blob/ed9e1c24622747aecd6767b01c70a818406948de/src/sgp/SGP.cc#L601

I prefer a more search friendly pattern:
```cpp
ARMY_COMPOSITION tmp = ARMY_COMPOSITION{};// or ARMY_COMPOSITION tmp{};
tmp.iReadability = compositionId;
tmp.bPriority = priority;
tmp.bElitePercentage = elitePercentage;
tmp.bTroopPercentage = troopPercentage;
tmp.bAdminPercentage = adminPercentage;
tmp.bDesiredPopulation = desiredPopulation;
tmp.bStartPopulation = startPopulation;
return tmp;
```

Reply:
I totally understand your point ([#1017](https://github.com/ja2-stracciatella/ja2-stracciatella/pull/1017/files) 🤦‍♂️).

But can still use `const` class member with zero-initialize constructor? The class data should be read-only.

Reply:
`ARMY_COMPOSITION ArmyCompositionModel::toArmyComposition() const`

This const means you are will not change the contents of ArmyCompositionModel.
You are only reading the members to fill a different structure, so there is no problem.